src/query/plan/operator.cpp

```diff
@@ -4695,10 +4697,120 @@ class CallProcedureCursor : public Cursor {
   }
 };
 
+class CallValidateProcedureCursor : public Cursor {
+  const CallProcedure *self_;
+  UniqueCursorPtr input_cursor_;
+
+ public:
+  CallValidateProcedureCursor(const CallProcedure *self, utils::MemoryResource *mem)
+      : self_(self), input_cursor_(self_->input_->MakeCursor(mem)) {}
+
+  bool Pull(Frame &frame, ExecutionContext &context) override {
+    SCOPED_PROFILE_OP("CallValidateProcedureCursor");
+
+    if (MustAbort(context)) throw HintedAbortError();
+    if (!input_cursor_->Pull(frame, context)) {
+      return false;
+    }
+
+    ExpressionEvaluator evaluator(&frame, context.symbol_table, context.evaluation_context, context.db_accessor,
+                                  storage::View::NEW);
+
+    const auto args = self_->arguments_;
+    MG_ASSERT(args.size() == 3);
+
+    const auto predicate = args[0]->Accept(evaluator);
+    const bool predicate_val = predicate.ValueBool();
+
+    if (predicate_val) [[unlikely]] {
+      const auto &message = args[1]->Accept(evaluator);
+      const auto &message_args = args[2]->Accept(evaluator);
+
+      auto format_error_message = [message_val = message.ValueString(),
+                                   message_args_val = message_args.ValueList()]() mutable {
+        std::size_t found{0U};
+        std::size_t arg_index{0U};
+
+        if (message_val.find('%', found) == std::string::npos) {
+          return message_val;
+        }
+
+        while (true) {
+          found = message_val.find('%', found);
+          if (found == std::string::npos) {
+            break;
+          }
+
+          // If someone finishes the message with '%' that could be error prone.
+          if (found == message_val.size() - 1U) {
+            break;
+          }
+
+          std::string replacement_str;
+          const auto format_specifier = message_val.at(found + 1U);
+          if (!std::isalpha(format_specifier)) {
+            ++found;
+            continue;
+          }
+          const bool does_argument_list_overflow = (message_args_val.size() < arg_index - 1U) && (arg_index > 0U);
```

Comment:
I think this is the correct order. Take this string as an example,
`"The test with test id:%d has been completed by student with student id:%d with 96% accuracy."[2,123]`
If we were to switch the order of the checks we would throw for this otherwise correct format string.